internal/staticupload/staticupload.go

```diff
@@ -57,6 +62,14 @@ type Config struct {
 	CacheInvalidationWaitTimeout time.Duration
 }
 
+// Validate checks if all necessary values are set and sets default values otherwise.
+func (c *Config) Validate() error {
+	if c.DistributionID == "" {
+		c.DistributionID = DefaultDistributionID
+	}
+	return nil
+}
```

Comment:
```suggestion
// SetDefaults sets default values.
func (c *Config) SetDefaults() error {
	if c.DistributionID == "" {
		c.DistributionID = DefaultDistributionID
	}
	return nil
}
```

nit: i expect a `Validate` method to never change the underlying data.